File: VienCNTT/BaiGiang/App_Code/Gio.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

/// <summary>
/// Summary description for Gio
/// </summary>
public class Gio
{
    //field:
    List<Hang> dshh = new List<Hang>();
    public double TongTien
    {
        get
        {
            var tong = 0.0;
            foreach (Hang h in this.dshh)
                tong += h.Gia * h.SoLuong;
            return tong;
        }
    }
    public int SoMatHang
    {
        get
        {
            return this.dshh.Count;
        }
    }
    private Hang Tim(int ma)
    {
        foreach (Hang h in this.dshh)
            if (h.Ma == ma) return h;
        return null;
    }
    public void Mua(Hang h, int sl)
    {
        var kq = Tim(h.Ma);
        if (kq == null)//them moi
        {
            h.SoLuong = sl;
            this.dshh.Add(h);
        }
        else//cap nhat so luong
        {
            kq.SoLuong += sl;
        }
    }
    public override string ToString()
    {
        string kq = string.Empty;
        foreach (Hang h in this.dshh)
            kq = kq + "<p>" + h.ToString() + "</p>";
        return kq;
    }
    public void Xoa(int ma)
    {
        var kq = Tim(ma);
        if (kq != null) this.dshh.Remove(kq);
    }
    public void Xoa()//xoa het hang hoa trong gio
    {
        this.dshh.Clear();
    }
    //viet property tong so luong hang hoa trong gio:
    public int TongSoLuong
    {
        get
        {
            var tong = 0;
            foreach (Hang h in this.dshh)
                tong += h.SoLuong;
            HttpContext.Current.Response.Write("ben trong = " + tong);
            return tong;
        }
    }
}

File: VienCNTT/BaiGiang/BH9/Gio.js
function Gio() {
    //mang chua ds hang hoa:
    this.ds = new Array();
    this.tim = function (ma) {
        for (var i = 0; i < this.ds.length; i++)
            if (ma == this.ds[i].ma) return this.ds[i];
        return null;
    }
    this.mua = function (h, n) {
        var kq = this.tim(h.ma);
        if (kq == null)//them moi:
        {
            var kt = this.ds.length++;//kt = gia tri cua length truoc khi ++
            h.soLuong = n;
            this.ds[kt] = h;
        }
        else//cap nhat so luong
            kq.soLuong += n;
    };
    this.tongTien = function () {
        var s = 0;
        for (var i = 0; i < this.ds.length; i++)
            s += this.ds[i].gia * this.ds[i].soLuong;
        return s;
    };
    this.xoa = function (ma) {
        //tim vi tri hang hoa can xoa:
        var vt = -1;
        for(var i=0;i<this.ds.length;i++)
            if (this.ds[i].ma == ma) {
                vt = i;
                break;
            }
        //xoa phan tu o vt bang cach chuyen du lieu & xoa pt cuoi cung:
        if (vt != -1) {
            for (var i = vt; i < this.ds.length - 1; i++)
                this.ds[i] = this.ds[i + 1];
            this.ds.length--;
        }
    }
}